### PR TITLE
dialect/sql/schema: support `time with time zone` and `time without time zone`

### LINKED ISSUE #2244: Postgres dialect should support 'time without time zone' type
Hi

I'm working on a project where I'm trying to store date and time in a timezone independent manner so i'm using the postgres `time` and `date` types via `SchemaType`. However I get the error `invalid type "time without time zone" for coumn...` when running migrations.

```go
func (FlightSchedule) Fields() []ent.Field {
	return []ent.Field{
		field.UUID("id", uuid.UUID{}).Default(uuid.New).Unique().Immutable(),
		field.Enum("weekday").GoType(enums.WeekDay(0)).Optional(),
		field.Enum("schedule_type").GoType(enums.FlightScheduleType("")),
		field.String("custom_date").GoType(customtypes.Date{}).SchemaType(map[string]string{dialect.Postgres: "date"}).Optional(),
		field.String("departs_at").GoType(customtypes.Time{}).SchemaType(map[string]string{dialect.Postgres: "time"}),
		field.String("arrives_at").GoType(customtypes.Time{}).SchemaType(map[string]string{dialect.Postgres: "time"}),
		field.Time("created_at").Default(time.Now),
		field.Time("updated_at").Default(time.Now).UpdateDefault(time.Now),
	}
}
```
I figured the error occurs because postgres `time` type is equivalent to and represented as `time without time zone` and this is not accounted for in the [scanColumn](https://github.com/ent/ent/blob/0c4e9c78a71a91544ec1e4b8b96d03a41dc1ff4e/dialect/sql/schema/postgres.go#L269) method. I believe adding `time without time zone` to the case statement will fix this issue.

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.16  |
| Ent         | 0.9.1  |
| Database    | Postgres   |
| Driver      | https://github.com/lib/pq |

### PR DESCRIPTION
**Pull request by s-takehana**
Fix #2244

8.5. Date/Time Types
https://www.postgresql.org/docs/14/datatype-datetime.html

### CODE REVIEW

File: dialect/sql/schema/postgres.go [hunk 1]
```diff
@@ -272,7 +272,7 @@ func (d *Postgres) scanColumn(c *Column, rows *sql.Rows) error {
 			schemaType := fmt.Sprintf("varchar(%d)", characterMaximumLen.Int64)
 			c.SchemaType = map[string]string{dialect.Postgres: schemaType}
 		}
-	case "date", "time", "timestamp", "timestamp with time zone", "timestamp without time zone":
+	case "date", "time with time zone", "time without time zone", "timestamp with time zone", "timestamp without time zone":
```
Comment (a8m) [inline dialect/sql/schema/postgres.go:275]:
```suggestion
	case "date", "time", "timestamp", "time with time zone", "time without time zone", "timestamp with time zone", "timestamp without time zone":
```

Please add back the "time" and "timestamp" types as they are also valid inputs.

**Review by a8m — APPROVED** [review]
Thanks for the contribution @s-takehana 

Looks good! Please, fix the minor comment and I'll merge this.

File: dialect/sql/schema/postgres.go (hunk 1 above)
Comment (s-takehana) [reply dialect/sql/schema/postgres.go:275]: Thank you for your review.
Is it necessary?

```go
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.Int("age").Positive(),
		field.String("name").Default("unknown"),
		field.Time("field1").Optional().
			SchemaType(map[string]string{
				dialect.Postgres: "time",
			}),
		field.Time("field2").Optional().
			SchemaType(map[string]string{
				dialect.Postgres: "timestamp",
			}),
		field.Time("field3").Optional().
			SchemaType(map[string]string{
				dialect.Postgres: "time with time zone",
			}),
		field.Time("field4").Optional().
			SchemaType(map[string]string{
				dialect.Postgres: "time without time zone",
			}),
	}
}
```

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id")) args=[]
```

```sql
postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```

`time` ➡️ `time without time zone`
`timestamp` ➡️ `timestamp without time zone`

Comment (a8m) [reply dialect/sql/schema/postgres.go:275]: Can you verify this on Postgres 11, 12, 13, and 14?

I don't want to bother you, but I usually double-check myself before removing some code. Thanks

Comment (s-takehana) [reply dialect/sql/schema/postgres.go:275]: I've confirmed it.

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id"))
```

```sql
postgres=# SHOW server_version_num;
 server_version_num
--------------------
 110014
(1 row)

postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id"))
```

```sql
postgres=# SHOW server_version_num;
 server_version_num
--------------------
 120009
(1 row)

postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id"))
```

```sql
postgres=# SHOW server_version_num;
 server_version_num
--------------------
 130005
(1 row)

postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id"))
```

```sql
postgres=# SHOW server_version_num;
 server_version_num
--------------------
 140001
(1 row)

postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```

**Comment by a8m** [comment]
Great! Thanks @s-takehana for the contribution 🙏

**Pull request merged**